File: app/src/main/java/com/esoxjem/movieguide/listing/sorting/SortingModule.java
package com.esoxjem.movieguide.listing.sorting;

import dagger.Module;
import dagger.Provides;

/**
 * @author pulkitkumar
 * @author arunsasidharan
 */
@Module
public class SortingModule {
    @Provides
    SortingDialogInteractor providesSortingDialogInteractor(SortingOptionStore store) {
        return new SortingDialogInteractorImpl(store);
    }

    @Provides
    SortingDialogPresenter providePresenter(SortingDialogInteractor interactor) {
        return new SortingDialogPresenterImpl(interactor);
    }
}

File: app/src/main/java/com/esoxjem/movieguide/movies/sorting/SortingModule.kt
package com.esoxjem.movieguide.movies.sorting

import dagger.Module
import dagger.Provides

/**
 * @author pulkitkumar
 * @author arunsasidharan
 */
@Module
class SortingModule {
    @Provides
    fun providesSortingDialogInteractor(store: SortingOptionStore): SortingContract.Interactor {
        return SortingDialogInteractor(store)
    }

    @Provides
    fun providePresenter(interactor: SortingContract.Interactor): SortingContract.Presenter {
        return SortingDialogPresenter(interactor)
    }
}
